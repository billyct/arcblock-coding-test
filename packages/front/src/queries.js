export const blockQuery = (hash) => `
{
  block(hash: ${JSON.stringify(hash)}) {
    hash
    height
    size
    weight
    time
    bits
    fee
    reward
    nonce
    mrklRoot
    version
    transactionsCount
  }
}
`

export const transactionsQuery = (hash, page) => `
{
  block(hash: ${JSON.stringify(hash)}) {
    transactions(page: ${page}) {
      hash
      time
      fee
      inputsCount
      inputsValueCount
      inputs {
        ... on TransactionInput {
          addr
          value
          spent
        }
        
        ... on CoinbaseInput {
          isCoinbaseInput
        }
      }
      outputsCount
      outputsValueCount
      outputs {
        ... on TransactionOutput {
          addr
          value
          spent
        }
        
        ... on OpReturnOutput {
          isOpReturnOutput
        }
      }
    }
  }
}
`
