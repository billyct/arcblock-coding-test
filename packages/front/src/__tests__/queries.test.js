import {transactionsQuery, blockQuery} from '../queries'

describe('test queries', () => {
  test('blockQuery', () => {
    expect(blockQuery(0)).toMatchSnapshot()
  })

  test('transactionsQuery', () => {
    expect(transactionsQuery(0, 1)).toMatchSnapshot()
  })
})
