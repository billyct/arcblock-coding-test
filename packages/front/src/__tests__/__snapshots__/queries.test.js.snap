// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test queries blockQuery 1`] = `
"
query GetBlock_0 {
  block(hash: 0) {
    hash
    height
    size
    weight
    time
    bits
    fee
    reward
    nonce
    mrklRoot
    version
    transactionsCount
  }
}
"
`;

exports[`test queries transactionsQuery 1`] = `
"
query GetTransactions_0_1 {
  block(hash: 0) {
    transactions(page: 1) {
      hash
      time
      fee
      inputsCount
      inputsValueCount
      inputs {
        ... on TransactionInput {
          addr
          value
          spent
        }
        
        ... on CoinbaseInput {
          isCoinbaseInput
        }
      }
      outputsCount
      outputsValueCount
      outputs {
        ... on TransactionOutput {
          addr
          value
          spent
        }
        
        ... on OpReturnOutput {
          isOpReturnOutput
        }
      }
    }
  }
}
"
`;
